src/base/QXmppSasl.cpp

```diff
@@ -923,18 +923,15 @@ QMap<QByteArray, QByteArray> QXmppSaslDigestMd5::parseMessage(const QByteArray &
 QByteArray QXmppSaslDigestMd5::serializeMessage(const QMap<QByteArray, QByteArray> &map)
 {
     QByteArray ba;
-    foreach (const QByteArray &key, map.keys())
-    {
+    for (const auto &key : map.keys()) {
```

Comment:
This allocates a whole list of keys and then does a lookup for each key a few lines later. Perhaps it'd be better to do old-style iterator-based `for` loop (or, alternatively, I have a handy view class for exactly this case, but it requires something between C++14 and 17).

Reply:
What about `std::for_each`?